parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -234,18 +237,53 @@ public Encoding getEncoding(org.apache.parquet.column.Encoding encoding) {
     return Encoding.valueOf(encoding.name());
   }
 
-  public static Statistics toParquetStatistics(org.apache.parquet.column.statistics.Statistics statistics) {
+  /**
+   * @deprecated Replaced by {@link #toParquetStatistics(
+   * String createdBy, org.apache.parquet.column.statistics.Statistics statistics)}
+   */
+  @Deprecated
+  public static Statistics toParquetStatistics(
+      org.apache.parquet.column.statistics.Statistics statistics) {
+    return toParquetStatistics(null, statistics);
+  }
+
+  public static Statistics toParquetStatistics(
+      String createdBy, org.apache.parquet.column.statistics.Statistics statistics) {
     Statistics stats = new Statistics();
     if (!statistics.isEmpty()) {
       stats.setNull_count(statistics.getNumNulls());
-      if(statistics.hasNonNullValue()) {
+      if (statistics.hasNonNullValue() &&
+          createdBy != null &&
+          !shouldIgnoreStatistics(createdBy)) {
         stats.setMax(statistics.getMaxBytes());
         stats.setMin(statistics.getMinBytes());
-     }
+      }
     }
     return stats;
   }
 
+  private static boolean shouldIgnoreStatistics(String createdBy) {
+    if (!createdBy.startsWith("parquet-mr")) {
+      return false; // Assume that other implementations got the statistics correct
+    }
+
+    return parseVersion(createdBy) < STATISTICS_FIXED_VERSION;
+  }
+
+  private static int parseVersion(String createdBy) {
+    final String[] versionTokens = createdBy.split(" ");
+
+    if (versionTokens.length < 3) {
+      return -1; // Ignore statistics
+    }
+
+    try {
+      return Integer.parseInt(versionTokens[2].substring(0, 5).replaceAll("\\.", ""));
```

Comment:
This parsing seems pretty brittle and doesn't account for changes in the format or for versions like 1.6.0rc2. What about using a regex (with tests) instead?